pydm/widgets/analog_indicator.py

```diff
@@ -0,0 +1,1576 @@
+
+        upper_major_alarm_start = self.calculate_position_for_value(self._upper_major_alarm)
+        upper_major_alarm_width = self._widget_width - upper_major_alarm_start
+
+        major_alarm_height = self._bg_size_rate * self._widget_height
+
+        """
+        sets the pen color to alarm if the value is in the lower major alarm region
+        """
+        if self._value < self._lower_major_alarm:
+            self._painter.setBrush(self._major_alarm_color)
+        else:
+            self._painter.setBrush(self._major_alarm_region_color)
+
+
+        self._painter.drawRect(0, 15, lower_major_alarm_width, major_alarm_height)
+
+        """
+        sets the pen color to alarm if the value is in the upper major alarm region
+        """
+        if self._value > self._upper_major_alarm:
+            self._painter.setBrush(self._major_alarm_color)
+        else:
+            self._painter.setBrush(self._major_alarm_region_color)
+
+        self._painter.drawRect(upper_major_alarm_start, 15, upper_major_alarm_width, major_alarm_height)
+
+    def paintEvent(self, event):
+        """
+        Paint events are sent to widgets that need to update themselves,
+        for instance when part of a widget is exposed because a covering
+        widget was moved.
+
+        Parameters
+        ----------
+        event : QPaintEvent
+        """
+        self.adjust_transformation()
+        self._painter.begin(self)
+        self._painter.translate(0, self._painter_translation_y) # Draw vertically if needed
+        self._painter.rotate(self._painter_rotation)
+        self._painter.translate(self._painter_translation_x, 0) # Invert appearance if needed
+        self._painter.scale(self._painter_scale_x, 1)
+
+        self._painter.translate(0, self._flip_traslation_y)     # Invert scale if needed
+        self._painter.scale(1, self._flip_scale_y)
+
+        self._painter.setRenderHint(QPainter.Antialiasing)
+
+        """
+        self.draw_indicator() -> draws a triangle outside of the scale, used as the value indicator
+        self.draw_background() -> draws the background of the scale
+        self.draw_major() -> draws the major alarm area
+        self.draw_minor() -> draws the minor alarm area
+        self.draw_ticks() -> draws scale ticks
+        self.draw_set_point -> draws diamond in the middle of the scale. can use current self.draw_indicator() for this
+        """
+        self.draw_background()
+        self.draw_ticks()
+
+        self.draw_minor_alarm_region()
+        self.draw_major_alarm_region()
+        self.draw_indicator()
+        #self.draw_set_point()
+
+        self._painter.end()
+
+    def calculate_position_for_value(self, value):
+        """
+        Calculate the position (pixel) in which the pointer should be drawn for a given value.
+        """
+        if value is None or value < self._lower_limit or value > self._upper_limit or \
```

Comment:
This may give strange results when a user mistakenly puts in lower limit as the high value and upper limit as the low value. Perhaps that should be handled here, or maybe there should be some input validation on the upper/lower limits setters.